scim2-sdk-server/src/test/java/com/unboundid/scim2/server/EndpointTestCase.java

```diff
@@ -1130,6 +1130,27 @@ public void testBadErrorResult() throws Exception
       Assert.assertEquals(response.getStatus(), Integer.valueOf(409));
       Assert.assertNotNull(ex.getCause());
     }
+
+    // ensure that for a failed request, if the returned entity could not
+    // be deserialized as an ErrorResponse, the SCIM exception contains a
+    // message and a fallback ErrorResponse describing the request failure
+    // NOT the deserialization failure - see DS-44767
```

Comment:
This comment and the fix itself actually apply to both cases exercised by this test method. Why don't you move it up to the top of the method or update the Javadoc?

I'd also update both cases so that they perform the same checks. E.g.:

* ErrorResponse.getStatus returns the expected HTTP status code
* ScimServiceException.getCause does not return a null
* ScimServiceException.getMessage equals ErrorResponse.getDetail
* ErrorResponse.getDetail returns the expected value

I suppose you could even parameterize the test to ensure consistency if you ever want to add other test cases.